pkg/controller/command/command_upgrade.go

```diff
@@ -30,8 +49,156 @@ func performUpgradeStepIfNeeded(commandCR *contrail.Command, currentDeployment *
+	err := r.client.Get(context.Background(), jobName, dataMigrationJob)
+	alreadyExists := err == nil
+	if alreadyExists {
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	dataMigrationJob = r.dataMigrationJob(command, oldImage, newImage, configMapName)
+	if err = controllerutil.SetControllerReference(command, dataMigrationJob, r.scheme); err != nil {
+		return err
+	}
+	return r.client.Create(context.Background(), dataMigrationJob)
+}
+
+func (r *ReconcileCommand) dataMigrationJob(commandCR *contrail.Command, oldImage, newImage, configMapName string) *batch.Job {
+	executableMode := int32(0744)
+	return &batch.Job{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      commandCR.Name + "-upgrade-job",
+			Namespace: commandCR.Namespace,
+		},
+		Spec: batch.JobSpec{
+			BackoffLimit: int32ToPtr(5),
+			Template: core.PodTemplateSpec{
+				Spec: core.PodSpec{
+					HostNetwork:   true,
+					RestartPolicy: core.RestartPolicyNever,
+					NodeSelector:  commandCR.Spec.CommonConfiguration.NodeSelector,
+					Volumes: []core.Volume{
+						{
+							Name: configMapName,
+							VolumeSource: core.VolumeSource{
+								ConfigMap: &core.ConfigMapVolumeSource{
+									LocalObjectReference: core.LocalObjectReference{
+										Name: configMapName,
+									},
+									DefaultMode: &executableMode,
+								},
+							},
+						},
+						{
+							Name: "backup-volume",
+							VolumeSource: core.VolumeSource{
+								EmptyDir: &core.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					InitContainers: []core.Container{
+						{
+							Name:            "command-upgrade-db-dump",
+							ImagePullPolicy: core.PullAlways,
+							Image:           oldImage,
+							Command:         []string{"bash", "-c", dataDumpScript},
+							VolumeMounts: []core.VolumeMount{{
+								Name:      configMapName,
+								MountPath: "/etc/contrail",
+							}, {
+								Name:      "backup-volume",
+								MountPath: "/backups/",
+							}},
+						},
+					},
+					Containers: []core.Container{
+						{
+							Name:            "command-upgrade-migrate",
+							ImagePullPolicy: core.PullAlways,
+							Image:           newImage,
+							Command:         []string{"bash", "-c", dataMigrationScript},
+							VolumeMounts: []core.VolumeMount{{
+								Name:      configMapName,
+								MountPath: "/etc/contrail",
+							}, {
+								Name:      "backup-volume",
+								MountPath: "/backups/",
+							}},
+						},
+					},
+					DNSPolicy: core.DNSClusterFirst,
+					Tolerations: []core.Toleration{
+						{Operator: "Exists", Effect: "NoSchedule"},
+						{Operator: "Exists", Effect: "NoExecute"},
+					},
+				},
+			},
+			TTLSecondsAfterFinished: nil,
+		},
+	}
+}
+
+const dataDumpScript = `
+#!/bin/bash
+
+commandutil convert --intype rdbms --outtype yaml --out /backups/db.yml -c /etc/contrail/command-app-server.yml
+`
+
+const dataMigrationScript = `
+#!/bin/bash
+
+commandutil migrate --in /backups/db.yml --out /backups/db_migrated.yml
```

Comment:
What happen when first command fail? Will the next command will be executed?

Reply:
What do you think about having a single command per one container (2 init containers and 1 container)?

Reply:
changed